include/storage/fbw.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for flash buffered writes
+ */
+
+#ifndef ZEPHYR_INCLUDE_STORAGE_FBW_H_
+#define ZEPHYR_INCLUDE_STORAGE_FBW_H_
+
+/**
+ * @brief Abstraction over flash buffered writes.
+ *
+ * @defgroup fbw flash buffered write Interface
+ * @{
+ */
+
+#include <stdbool.h>
+#include <storage/flash_map.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct fbw_ctx {
+	u8_t *buf;
+	size_t buf_len;
+	const struct flash_area *flash_area;
+	size_t bytes_written;
+	u16_t buf_bytes;
+#ifdef CONFIG_IMG_ERASE_PROGRESSIVELY
```

Comment:
This is a flash_img.h config option; right? I think it shouldn't be in this API if so. Would it need to get renamed as part of making this API generic?

Reply:
Yes, I made a separate Kconfig for this module, will change to this.

Reply:
CONFIG_FBW_ERASE_PROGRESSIVELY
Wuld be great if IMG_ERASE_PROGRESSIVELY will be kept as selector for the  new keyword - this will ensure API backward compatibility.

Reply:
I planned on keeping the backward compatibility by having IMG_ERASE_PROGRESSIVELY select FBW_ERASE_PROGRESSIVELY